src/jbtoc.ts

```diff
@@ -0,0 +1,315 @@
+import { ServerConnection } from '@jupyterlab/services';
+import * as path from 'path';
+import * as yaml from 'js-yaml';
+
+interface IFileMetadata {
+  path: string;
+}
+
+interface IJbookConfig {
+  title: string;
+  author: string;
+  logo: string;
+}
+
+interface IToc {
+  parts?: IPart[];
+  chapters?: ISection[];
+  caption?: string;
+}
+
+interface ISection {
+  sections?: ISection[];
+  file?: string;
+  url?: string;
+  title?: string;
+  glob?: string;
+}
+
+interface IPart {
+  caption: string;
+  chapters: ISection[];
+}
```

Comment:
I'm working on lifting the ToC parsing / validation logic to a TS package in MyST-MD! I'll bring it in here once it's published.

Reply:
That's excellent! Will it parse a toc yaml object that you pass it or will it also handle needed file system operations? 

This latest version (0.2.0) isn't quite working in JupyterLite yet because I'm ls'ing by hitting the contents API but I think I can do that with jupyterlab/services.ContentManager. File loading is already handled with the jupyterlab/services API.